### PR TITLE
[AKS] refactor to use Knack framework 

### PR DESCRIPTION
**Pull request by mboersma**
Includes changes from #4996 and #5029.

### CODE REVIEW

**Comment by derekbekoe** [comment]
@mboersma I don't think we have CI green on this branch yet.

Run `check_style --module acs && run_tests --module acs` locally and verify all pass.
Also, prompt.ws (in the comment above) can be used to verify. It take a couple minutes to load.

@tjprescott to confirm on CI being green?

**Comment by mboersma** [comment]
The `check_style` and `run_tests` do pass for the acs module. I added e2f9dec to fix the two flake8 errors it found in core. 🤞

**Comment by tjprescott** [comment]
Right, I'm working on making CI green now.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_actions.py [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
-        if public_key_filepath[-4:].lower() == '.pub':
-            private_key_filepath = public_key_filepath[:-4]
-        else:
-            private_key_filepath = public_key_filepath + '.private'
-        content = generate_ssh_keys(private_key_filepath, public_key_filepath)
-        logger.warning('Created SSH key files: %s,%s', private_key_filepath, public_key_filepath)
-    args.ssh_key_value = content
-
-# TODO: Move this into the AcsCommandsLoader file...
-#AZ_CLI.register(AZ_CLI.COMMAND_PARSER_PARSED, _handle_container_ssh_file)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_actions.py]: Where did you move this to?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_completers.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from azure.cli.core.decorators import Completer
+
+from ._client_factory import cf_compute_service
+
+
+@Completer
+def get_vm_size_completion_list(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    print(prefix)
+    print(namespace)
+    raise NotImplementedError()
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_completers.py]: Why is this here?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 3]
```diff
@@ -3,46 +3,154 @@
...
+import os.path
 import platform
 
 from argcomplete.completers import FilesCompleter
-
-from azure.cli.core.commands import register_cli_argument, register_extra_cli_argument
+from azure.cli.core.commands.parameters import file_type
+from azure.cli.core.commands.parameters import get_enum_type
+from azure.cli.core.commands.parameters import get_resource_name_completion_list
+from azure.cli.core.commands.parameters import name_type
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: These can all be consolidated into a single line
```Python
from azure...parameters import (
  file_type, name_type, get_enum_type, get_resource_name_completion_list)
```

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_client_factory.py [hunk 4]
```diff
@@ -3,42 +3,52 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+from azure.cli.core.commands.client_factory import configure_common_settings
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
 
-def _auth_client_factory(scope=None):
+
+def cf_compute_service(cli_ctx, _):
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_client_factory.py]: Change `_` to `*_` then you don't need to call it like `cf_compute_service(cli_ctx, None)` as you can just do `cf_compute_service(cli_ctx)`. Same with the others.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 5]
```diff
@@ -3,46 +3,154 @@
...
+        c.argument('tags', tags_type)
+        c.argument('disable_browser', help='Do not open browser after opening a proxy to the cluster web user interface')
+
+    with self.argument_context('acs create') as c:
+        c.argument('ssh_key_value', required=False, type=file_type, default=os.path.join('~', '.ssh', 'id_rsa.pub'),
+                   completer=FilesCompleter(), validator=validate_ssh_key)
+        c.argument('master_profile', options_list=['--master-profile', '-m'], type=validate_file_or_dict,
+                   help=_get_feature_in_preview_message() + 'The file or dictionary representation of the master profile. Note it will override any master settings once set')
+        c.argument('master_vm_size', completer=get_vm_size_completion_list, help=_get_feature_in_preview_message())
+        # c.argument('master_osdisk_size', type=int, )
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: unfinished?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 6]
```diff
@@ -60,124 +168,45 @@ def _get_default_install_location(exe_name):
...
+    "japanwest",
+    "northcentralus",
+    "northeurope",
+    "southcentralus",
+    "southeastasia",
+    "westeurope",
+    "westus",
+]
+
+storage_profile_types = ["StorageAccount", "ManagedDisks"]
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: Would prefer these consts. at the top of file after the imports.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py (hunk 3 above)
Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: I've always followed the [PEP8](https://www.python.org/dev/peps/pep-0008/#imports) recommendation of `Imports should be on separate lines`, but if this style is preferred locally, that's fine with me.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_completers.py (hunk 2 above)
Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_completers.py]: Oops, some debug code I failed to remove.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py (hunk 5 above)
Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: Just some cruft I'll remove. `master_osdisk_size` is properly defined a couple lines below.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/__init__.py [hunk 7]
```diff
@@ -4,12 +4,26 @@
...
 
 
+from azure.cli.core import AzCommandsLoader
+
 import azure.cli.command_modules.acs._help  # pylint: disable=unused-import
 
 
-def load_params(_):
-    import azure.cli.command_modules.acs._params  # pylint: disable=redefined-outer-name, unused-variable
+class ContainerCommandsLoader(AzCommandsLoader):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/__init__.py]: This name will conflict with the container module. Please rename this.
